cpp/src/slice2php/Main.cpp

```diff
@@ -131,11 +131,7 @@ CodeVisitor::visitClassDecl(const ClassDeclPtr& p)
 
         string type = getTypeVar(p);
         _out << sp << nl << "global " << type << ';';
-
-        _out << nl << "if(!isset(" << type << "))";
-        _out << sb;
         _out << nl << type << " = IcePHP_declareClass('" << scoped << "');";
-        _out << eb;
```

Comment:
I believe that this check for `isset` is unnecessary.
This block is wrapped in an `if` statement:
```
if (_classHistory.count(scoped) == 0)
{...
_classHistory.insert(scoped); // Avoid redundant declarations.
}
```
which should ensure that this code is only generated once.

Even if I'm wrong (which I don't think I am), there's nothing wrong with setting `type = IcePHP_declareClass` multiple times, it's just redundant is all.